scanpy/experimental/pp/_highly_variable_genes.py

```diff
@@ -90,6 +92,47 @@ def clac_clipped_res_sparse(gene: int, cell: int, value: np.float64) -> np.float
     return residuals
 
 
+def _calculate_res_dense_vectorized(
```

Comment:
Why is this called `dense` if it operates on sparse inputs?  Also why are the type hints `np.ndarray` then?  Also, why the extra dtype?  Can we guarantee that?